### PR TITLE
Add probe field in class Templates

### PR DESCRIPTION
**Pull request by samuelgarcia**
Add probe field in class Templates

### CODE REVIEW

**Comment by samuelgarcia** [comment]
@alejoe91 : ready on my side

File: src/spikeinterface/core/template.py [hunk 1]
```diff
@@ -24,6 +25,8 @@ class Templates:
         Array of channel IDs. If `None`, defaults to an array of increasing integers.
     unit_ids : np.ndarray, optional default: None
         Array of unit IDs. If `None`, defaults to an array of increasing integers.
+    probe: Probe , optional default: None
```
Comment (alejoe91) [inline src/spikeinterface/core/template.py:28]:
```suggestion
    probe: Probe, default: None
```

Comment (alejoe91) [reply src/spikeinterface/core/template.py:28]: can this be a `ProbeGroup` too?

**Comment by alejoe91** [comment]
@samuelgarcia pre-commit?

File: src/spikeinterface/core/template.py (hunk 1 above)
Comment (samuelgarcia) [reply src/spikeinterface/core/template.py:28]: No. This is too complicated. A template is Probe only.

**Comment by samuelgarcia** [comment]
yep

File: src/spikeinterface/core/template.py [hunk 2]
```diff
@@ -24,6 +25,8 @@ class Templates:
         Array of channel IDs. If `None`, defaults to an array of increasing integers.
     unit_ids : np.ndarray, optional default: None
         Array of unit IDs. If `None`, defaults to an array of increasing integers.
+    probe: Probe, default: None
+        A probeinterface.probe object
```
Comment (alejoe91) [inline src/spikeinterface/core/template.py:29]:
```suggestion
        A `probeinterface.Probe` object
```

**Review by alejoe91 — APPROVED** [review]

**Review by h-mayorquin — APPROVED** [review]

File: src/spikeinterface/core/tests/test_template_class.py [hunk 3]
```diff
@@ -15,6 +17,9 @@ def generate_test_template(template_type):
     sampling_frequency = 30_000
     nbefore = 2
 
+    probe = generate_multi_columns_probe(num_columns=1, num_contact_per_column=[3])
+    print(probe)
```
Comment (h-mayorquin) [inline src/spikeinterface/core/tests/test_template_class.py:21]:
```suggestion
```

**Pull request merged**